scripts/lib/install

```diff
@@ -205,6 +243,63 @@ case "$POSTGRESQL_VERSION" in
         ;;
 esac
 
+if [ -z "$PUSH_NOTIFICATIONS" ] && [ -z "$NO_PUSH_NOTIFICATIONS" ]; then
+    # Unless specified, we default to --no-push-notifications
+    NO_PUSH_NOTIFICATIONS=1
+fi
+
+# We set these to Python-style True/False string values, for easy
+# insertion into the settings.py file.
+if [ -n "$PUSH_NOTIFICATIONS" ]; then
+    SERVICE_MOBILE_PUSH="True"
+
+    # --push-notifications also enables SUBMIT_USAGE_STATISTICS, unless the user
+    # explicitly opted out by passing --no-submit-usage-statistics.
+    if [ -n "$NO_SUBMIT_USAGE_STATISTICS" ]; then
+        SERVICE_SUBMIT_USAGE_STATISTICS="False"
+    else
+        SERVICE_SUBMIT_USAGE_STATISTICS="True"
+    fi
+else
+    SERVICE_MOBILE_PUSH="False"
+    # SUBMIT_USAGE_STATISTICS without PUSH_NOTIFICATIONS wouldn't really
+    # make much sense.
+    SERVICE_SUBMIT_USAGE_STATISTICS="False"
+fi
+
+# If user asked for push notifications, prompt for ToS acceptance.
+if [ -n "$PUSH_NOTIFICATIONS" ]; then
+    echo
+    echo "You chose to register your server for the Mobile Push Notifications Service."
+    echo "Doing so will share basic metadata with the service's maintainers, including:"
+    echo
+    echo "* The server's configured hostname: $EXTERNAL_HOST"
+    echo "* The server's configured contact email address: $ZULIP_ADMINISTRATOR"
+    echo "* Basic metadata about each organization hosted by the server; see:"
+    echo "    <https://zulip.readthedocs.io/en/latest/production/mobile-push-notifications.html#uploading-basic-metadata>"
+    echo
+    echo "Use of this service is governed by the Zulip Terms of Service:"
+    echo "    <https://zulip.com/policies/terms>"
```

Comment:
Do we want to link to https://zulip.readthedocs.io/en/latest/production/mobile-push-notifications.html#why-a-push-notification-service-is-necessary or something, so folks can understand why this is necessary at all?